dmn-engine/src/main/scala/org/camunda/dmn/evaluation/ContextEvaluator.scala

```diff
@@ -1,29 +1,23 @@
 package org.camunda.dmn.evaluation
 
-import scala.collection.JavaConverters._
-
+import org.camunda.dmn.Audit.ContextEvaluationResult
 import org.camunda.dmn.DmnEngine._
 import org.camunda.dmn.FunctionalHelper._
-import org.camunda.feel.FeelEngine
-import org.camunda.feel.context.Context._
-import org.camunda.bpm.model.dmn.instance.{Context, ContextEntry, Expression}
 import org.camunda.dmn.parser.{ParsedContext, ParsedDecisionLogic}
+import org.camunda.feel.context.Context._
 import org.camunda.feel.syntaxtree.{Val, ValContext, ValFunction}
-import org.camunda.dmn.Audit.ContextEvaluationResult
 
 class ContextEvaluator(
     eval: (ParsedDecisionLogic, EvalContext) => Either[Failure, Val]) {
 
   def eval(context: ParsedContext, ctx: EvalContext): Either[Failure, Val] = {
 
-    evalContextEntries(context.entries, ctx).right.flatMap(results =>
-      evalContextResult(context.aggregationEntry, results, ctx).right.map {
-        result =>
-          ctx.audit(context,
-                    ContextEvaluationResult(entries = results, result = result))
-
-          result
-    })
+    evalContextEntries(context.entries, ctx).flatMap(
+      results =>
+        ctx.audit(evalContextResult(context.aggregationEntry, results, ctx),
+                  context,
+                  (v: Val) =>
+                    ContextEvaluationResult(entries = results, result = v)))
```

Comment:
With the suggested audit function, this could look like: 

```suggestion
    evalContextEntries(context.entries, ctx).flatMap { results =>
      val result = evalContextResult(context.aggregationEntry, results, ctx)

      ctx.audit(context, result, r => ContextEvaluationResult(entries = results, result = r))
      result
    }
```

I think it is easier to read and makes the audit and the return value more clear.